lightning/src/offers/invoice.rs

```diff
@@ -571,12 +581,28 @@ type BlindedPayInfoIter<'a> = core::iter::Map<
 /// Information needed to route a payment across a [`BlindedPath`].
 #[derive(Clone, Debug, PartialEq)]
 pub struct BlindedPayInfo {
-	fee_base_msat: u32,
-	fee_proportional_millionths: u32,
-	cltv_expiry_delta: u16,
-	htlc_minimum_msat: u64,
-	htlc_maximum_msat: u64,
-	features: BlindedHopFeatures,
+	/// Base fee charged (in millisatoshi) for the entire blinded path.
+	pub fee_base_msat: u32,
+
+	/// Liquidity fee charged (in millionths of a satoshi) for the entire blinded path (i.e., 10,000
+	/// is 1%).
+	pub fee_proportional_millionths: u32,
+
+	/// Number of blocks subtracted from an incoming HTLC's `cltv_expiry` for the entire blinded
+	/// path.
+	pub cltv_expiry_delta: u16,
+
+	/// The minimum HTLC value (in millisatoshi) that is acceptable to all channel peers on the
```

Comment:
Hmm... the blinded path is treated as one hop with regard to path finding. So you can think of this being the minimum amount the final node is willing to accept over the "hop" from the introduction point, which would mean that it must be acceptable to all peers on the path, no? (e.g., if the final hop is 1 msat and one of the previous hops is 10 msats, this value would need to be 10 msats. i.e., it's the maximum of all minimums)

Maybe I don't follow your question though.

Reply:
Gotcha. Adjusted the comment to note fees along the blinded path need to be accounted for.

Reply:
IIUC, the previous hop should be for the amount seen by the introduction point since that is the hop *to* it. So this must therefore be for the amount seen by the final node.

Updated. Let me know if I misunderstood or if you have a different wording in mind.